Change columns order and visibility in DrILL

gui-co opened the pull request:
**Description of work.**

This PR adds two features in the table of the DrILL interface:
* the visibility of the columns can be toggled via the context menu
* the order of the columns can be changed by drag-and-drop

These two visual settings are also exported/imported to/from a 'rundex' file.

**To test:**

Open the DrILL interface and play with it:
* move the columns to change their order
* hide/show them through the context menu (mouse right click most of the time)
* save the 'rundex' file (*File* -> *Save rundex as*) and reload it (*File* -> *Load rundex*) to see if the table state is preserved

Part of #29570 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

gvardany reviewed (changes requested):
There is a problem when changing the instrument or acquisition mode after having removed some columns. It seems it remembers the indices of the removed ones which are no longer relevant if you change the technique. Also could you add the add/remove actions to the edit menu as well? It could also be grouped like Edit->[Rows>[Add, Copy, Paste,...] , Columns>[Add/Remove] ]. Also please don't forget to add a section about these new features in the documentation.

gui-co commented:
The table is now reset when the instrument and/or acquisition mode is selected.

I added a new menu in the menu bar to hide/show columns. I also reorganized the menu by adding "rows" and "columns" submenus.

I added a note in DrILL's doc about the new features.

gvardany commented:
Thanks for the changes, it works well. Just a minor point, you can also put the Erase selected cells menu item under Cell(s) to be consistent with rows and columns. Also make clear in the docs that the order, the hidden and folded columns will be preserved when saving/loading a rundex file, whereas when changing the instrument and/or the acquisition mode it always starts fresh.

gui-co commented:
I added the 'Cell(s)' submenu for consistency.
I also clarified the doc concerning the saving and loading of visual settings.

gvardany reviewed (commented):
There is still something confusing, when I switch to polarized for D17, the Run01, 10 and 11 columns show up at the end, but they are supposed to be in the beginning.

gui-co commented:
The visual settings (column order, collapsed columns, hidden columns) are now reset when the instrument and/or acquisition mode is changed.

gvardany commented:
The column order doesn't seem to be reset, try reordering and swapping between polarized and unpolarized reflectometries.

gui-co commented:
The column order was somehow kept in the table header. I now reset it each time the table is created.

gvardany reviewed (approved):
Works as described.
